src/Action.hs

```diff
@@ -171,26 +182,21 @@ currentUserId = userState usUserId >>= \case
     Nothing -> throwError500 "User is logged out"
     Just uid -> pure uid
 
-currentUserAddDb :: ( Acid.UpdateEvent (AUpdate a)
-                    , Acid.MethodState (AUpdate a) ~ AulaData
-                    , Acid.MethodResult (AUpdate a) ~ a
-                    , ActionPersist m, ActionUserHandler m) =>
-                    (UserWithProto a -> AUpdate a) -> Proto a -> m a
+currentUserAddDb :: (HasAUpdate ev a, ActionPersist m, ActionUserHandler m) =>
+                    (UserWithProto a -> ev) -> Proto a -> m a
 currentUserAddDb addA protoA = do
     cUser <- currentUser
     aupdate $ addA (cUser, protoA)
 
-currentUserAddDb_ :: ( Acid.UpdateEvent (AUpdate a)
-                     , Acid.MethodState (AUpdate a) ~ AulaData
-                     , Acid.MethodResult (AUpdate a) ~ a
-                     , ActionPersist m, ActionUserHandler m) =>
-                    (UserWithProto a -> AUpdate a) -> Proto a -> m ()
+currentUserAddDb_ :: (HasAUpdate ev a, ActionPersist m, ActionUserHandler m) =>
+                     (UserWithProto a -> ev) -> Proto a -> m ()
 currentUserAddDb_ addA protoA = void $ currentUserAddDb addA protoA
 
 -- | Returns the current user
 currentUser :: (ActionPersist m, ActionUserHandler m) => m User
 currentUser = do
-    muser <- aquery . findUser =<< currentUserId
+    uid <- currentUserId
+    muser <- aquery (findUser uid)
```

Comment:
wait, that's not login errors, it's one level down.  but @andorp explains to me that you are still right: current user should exist in the database, and if it doesn't, it's not persist's fault.